site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,264 @@
+| *tenant-id*      | yes       | *application-properties* | *string*     | Contains the ID of the tenant that is subject of the outcome of the operation. |
+| *status*         | yes       | *application-properties* | *int*        | Contains the status code indicating the outcome of the operation. Concrete values and their semantics are defined for each particular operation. |
+
+# Delivery States
+
+Hono uses the following AMQP message delivery states when receiving request messages from clients:
+
+| Delivery State | Description |
+| :------------- | :---------- |
+| *ACCEPTED*     | Indicates that Hono has successfully received and accepted the request for processing. |
+| *REJECTED*     | Indicates that Hono has received the request but was not able to process it. The *error* field contains information regarding the reason why. Clients should not try to re-send the request using the same message properties in this case. |
+
+# Tenant Format
+
+Most of the operations of the Tenant API allow or require the inclusion of tenant data in the payload of the
+request or response messages of the operation. Such payload is carried in the body of the corresponding AMQP 
+messages as part of a single *AMQP Value* section.
+
+The tenant data is carried in the payload as a UTF-8 encoded string representation of a single JSON object. It is an error to include payload that is not of this type.
+
+The table below provides an overview of the standard members defined for the JSON object:
+
+| Name                     | Mandatory | Type       | Default Value | Description |
+| :------------------------| :-------: | :--------- | :------------ | :---------- |
+| *tenant-id*              | *yes*     | *string*   |               | The ID of the tenant. |
+| *enabled*                | *no*      | *boolean*  | *true*        | If set to *false* the tenant is currently disabled.|
+| *adapterConfiguration*   | *yes*     | *array*    |               | A list of configuration options valid for certain adapters only. See [Adapter Configuration Format]({{< relref "#adapter-configuration-format" >}}) for details. **NB** This array must contain at least one element - an empty array is handled as an error. **NB** Only one entry with a matching adapter-id is tolerated. If multiple configurations for the same adapter-id are present it is handled as an error.|
+
+Additionally to the specified properties the JSON object MAY contain an arbitrary number of members with arbitrary names which can be of a scalar or a complex type. 
+This allows *clients* to add further information which might be e.g. relevant just for certain adapters only.
+
+### Adapter Configuration Format
+
+When providing / updating a tenant for each *Hono protocol adapter* a configuration must be provided.
+Protocol Adapters without any configuration will be assumed as disabled for that tenant.
```

Comment:
Why?

At least you should give a hint when a protocol adapter (or any other component) should check for these values.
Otherwise "assume" is not clear about "who" is assuming that.

I would rather like a solution where you CAN control certain protocol adapters
IMHO this gives only a collection

Reply:
I removed the first sentence requiring to give a configuration for each protocol adapter.
Furthermore i added a description that the protocol adapters are responsible for their configuration according to the properties set in the tenant api.

I'm not quite sure if we want to enforce this evaluation or if it is something optional. Any ideas?